### PR TITLE
Problem: client-cli wallet sync in integration test fail occasionally (fix #2093)

### LINKED ISSUE #2093: Problem: client-cli wallet sync in integration test fail occasionally
```
pytests/test_client_cli.py:224:
--
1768 | _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
1769 |  
1770 | self = <client_cli.Wallet object at 0x7fc42dee65e0>, force = False
1771 | disable_address_recovery = False, enable_fast_forward = True, batch_size = 20
1772 | block_height_ensure = 50
1773 | light_client_peers = '0000000000000000000000000000000000000000@127.0.0.1:26657,1000000000000000000000000000000000000000@127.0.0.1:26657'
1774 |  
1775 | def sync(
1776 | self,
1777 | force=False,
1778 | disable_address_recovery=False,
1779 | enable_fast_forward=True,
1780 | batch_size=20,
1781 | block_height_ensure=50,
1782 | light_client_peers="0000000000000000000000000000000000000000@127.0.0.1:26657,1000000000000000000000000000000000000000@127.0.0.1:26657"
1783 | ):
1784 | cmd = ["client-cli", "sync", "-n", self.name, "--batch-size", str(batch_size), "--block-height-ensure", str(block_height_ensure),
1785 | "--light-client-peers", light_client_peers]
1786 | if force:
1787 | cmd.append("--force")
1788 | if disable_address_recovery:
1789 | cmd.append("--disable-address-recovery")
1790 | if enable_fast_forward:
1791 | cmd.append("--enable-fast-forward")
1792 | args = self.auth_token
1793 | text = run(cmd, args)
1794 | if "Error" in text:
1795 | >           raise Exception(text)
1796 | E           Exception: Enter authentication token: Error: Tendermint RPC error: Error while calling tendermint RPC call
1797 | E            => Error response from tendermint RPC for request method (block_results) and params ([String("444")]): RPC error response: JsonRpcError { code: -32603, message: "Internal error", data: Some(String("could not find results for height #444")) }
1798 | E
1799 | E           Stack backtrace:
1800 | E              0: anyhow::private::new_adhoc
1801 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.32/src/backtrace.rs:10
1802 | E              1: client_common::tendermint::rpc_client::async_rpc_client::AsyncRpcClient::receive_response::{{closure}}
1803 | E                        at client-common/src/tendermint/rpc_client/async_rpc_client.rs:228
1804 | E              2: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
1805 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/future/mod.rs:66
1806 | E              3: core::future::poll_with_context
1807 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/future/mod.rs:84
1808 | E              4: client_common::tendermint::rpc_client::async_rpc_client::AsyncRpcClient::request::{{closure}}
1809 | E                        at client-common/src/tendermint/rpc_client/async_rpc_client.rs:96
1810 | E              5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
1811 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/future/mod.rs:66
1812 | E              6: core::future::poll_with_context
1813 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/future/mod.rs:84
1814 | E              7: client_common::tendermint::rpc_client::async_rpc_client::AsyncRpcClient::call::{{closure}}
1815 | E                        at client-common/src/tendermint/rpc_client/async_rpc_client.rs:141
1816 | E              8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
1817 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/future/mod.rs:66
1818 | E              9: core::future::poll_with_context
1819 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/future/mod.rs:84
1820 | E             10: client_common::tendermint::rpc_client::sync_rpc_client::SyncRpcClient::call::{{closure}}
1821 | E                        at client-common/src/tendermint/rpc_client/sync_rpc_client.rs:152
1822 | E             11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
1823 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/future/mod.rs:66
1824 | E             12: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
1825 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/core.rs:173
1826 | E             13: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
1827 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/loom/std/unsafe_cell.rs:14
1828 | E             14: tokio::runtime::task::core::Core<T,S>::poll
1829 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/core.rs:158
1830 | E             15: tokio::runtime::task::harness::Harness<T,S>::poll::{{closure}}
1831 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/harness.rs:113
1832 | E             16: core::ops::function::FnOnce::call_once
1833 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/ops/function.rs:232
1834 | E             17: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
1835 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:318
1836 | E             18: std::panicking::try::do_call
1837 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:331
1838 | E             19: __rust_try
1839 | E             20: std::panicking::try
1840 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:274
1841 | E             21: std::panic::catch_unwind
1842 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:394
1843 | E             22: tokio::runtime::task::harness::Harness<T,S>::poll
1844 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/harness.rs:89
1845 | E             23: tokio::runtime::task::raw::poll
1846 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/raw.rs:104
1847 | E             24: tokio::runtime::task::raw::RawTask::poll
1848 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/raw.rs:66
1849 | E             25: tokio::runtime::task::Notified<S>::run
1850 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/mod.rs:169
1851 | E             26: tokio::runtime::thread_pool::worker::Context::run_task::{{closure}}
1852 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/thread_pool/worker.rs:374
1853 | E             27: tokio::coop::budget::{{closure}}
1854 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/coop.rs:97
1855 | E             28: std::thread::local::LocalKey<T>::try_with
1856 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/local.rs:262
1857 | E             29: std::thread::local::LocalKey<T>::with
1858 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/local.rs:239
1859 | E             30: tokio::coop::budget
1860 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/coop.rs:79
1861 | E                 tokio::runtime::thread_pool::worker::Context::run_task
1862 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/thread_pool/worker.rs:352
1863 | E             31: tokio::runtime::thread_pool::worker::Context::run
1864 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/thread_pool/worker.rs:324
1865 | E             32: tokio::runtime::thread_pool::worker::run::{{closure}}
1866 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/thread_pool/worker.rs:309
1867 | E             33: tokio::macros::scoped_tls::ScopedKey<T>::set
1868 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/macros/scoped_tls.rs:64
1869 | E             34: tokio::runtime::thread_pool::worker::run
1870 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/thread_pool/worker.rs:306
1871 | E             35: tokio::runtime::thread_pool::worker::Launch::launch::{{closure}}
1872 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/thread_pool/worker.rs:285
1873 | E             36: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
1874 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/blocking/task.rs:38
1875 | E             37: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
1876 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/core.rs:173
1877 | E             38: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
1878 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/loom/std/unsafe_cell.rs:14
1879 | E             39: tokio::runtime::task::core::Core<T,S>::poll
1880 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/core.rs:158
1881 | E             40: tokio::runtime::task::harness::Harness<T,S>::poll::{{closure}}
1882 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/harness.rs:113
1883 | E             41: core::ops::function::FnOnce::call_once
1884 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/ops/function.rs:232
1885 | E             42: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
1886 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:318
1887 | E             43: std::panicking::try::do_call
1888 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:331
1889 | E             44: __rust_try
1890 | E             45: std::panicking::try
1891 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:274
1892 | E             46: std::panic::catch_unwind
1893 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:394
1894 | E             47: tokio::runtime::task::harness::Harness<T,S>::poll
1895 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/harness.rs:89
1896 | E             48: tokio::runtime::task::raw::poll
1897 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/raw.rs:104
1898 | E             49: tokio::runtime::task::raw::RawTask::poll
1899 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/raw.rs:66
1900 | E             50: tokio::runtime::task::Notified<S>::run
1901 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/task/mod.rs:169
1902 | E             51: tokio::runtime::blocking::pool::Inner::run
1903 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/blocking/pool.rs:230
1904 | E             52: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}::{{closure}}
1905 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/blocking/pool.rs:210
1906 | E             53: tokio::runtime::context::enter
1907 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/context.rs:72
1908 | E             54: tokio::runtime::handle::Handle::enter
1909 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/handle.rs:71
1910 | E             55: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
1911 | E                        at /drone/src/drone/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.20/src/runtime/blocking/pool.rs:209
1912 | E             56: std::sys_common::backtrace::__rust_begin_short_backtrace
1913 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys_common/backtrace.rs:130
1914 | E             57: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
1915 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/mod.rs:475
1916 | E             58: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
1917 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:318
1918 | E             59: std::panicking::try::do_call
1919 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:331
1920 | E             60: __rust_try
1921 | E             61: std::panicking::try
1922 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:274
1923 | E             62: std::panic::catch_unwind
1924 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:394
1925 | E             63: std::thread::Builder::spawn_unchecked::{{closure}}
1926 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/mod.rs:474
1927 | E             64: core::ops::function::FnOnce::call_once{{vtable.shim}}
1928 | E                        at /root/.rustup/toolchains/nightly-2020-04-07-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/ops/function.rs:232
1929 | E             65: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
1930 | E                        at /rustc/6dee5f1126dfd5c9314ee5ae9d9eb010e35ef257/src/liballoc/boxed.rs:1008
1931 | E             66: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
1932 | E                        at /rustc/6dee5f1126dfd5c9314ee5ae9d9eb010e35ef257/src/liballoc/boxed.rs:1008
1933 | E                 std::sys::unix::thread::Thread::new::thread_start
1934 | E                        at src/libstd/sys/unix/thread.rs:87
1935 | E             67: start_thread
1936 | E                        at /build/glibc-OTsEL5/glibc-2.27/nptl/pthread_create.c:463
1937 | E             68: __clone
1938 |  
1939 | bot/client_cli.py:295: Exception
```

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: add retrying

### CODE REVIEW

**Comment by tomtau** [comment]
bors try

File: client-core/src/wallet/syncer.rs [hunk 1]
```diff
@@ -565,9 +576,38 @@ impl<
             }
 
             // Fetch batch details if it cannot be fast forwarded
-            let blocks = self.env.client.block_batch(range.iter())?;
-            let block_results = self.env.client.block_results_batch(range.iter())?;
-            let states = self.env.client.query_state_batch(range.iter().cloned())?;
+            let mut blocks: Vec<Block>;
+            let mut block_results: Vec<BlockResultsResponse>;
+            let mut states: Vec<ChainState>;
+            // if any error occurs, do it again
```
Comment (tomtau) [inline client-core/src/wallet/syncer.rs:582]: should there be some limited number of retries?

Comment (leejw51crypto) [reply client-core/src/wallet/syncer.rs:582]: i'll add some limited number for retries

**Comment by leejw51crypto** [comment]
ok, i'll check

**Comment by leejw51crypto** [comment]
rebased

**Comment by tomtau** [comment]
bors retry

**Pull request merged**